### PR TITLE
Introduce build directory for artifacts

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
       - run:
           name: Test the contract
           command: |
-            cp /tmp/combined.json .
+            cp -r /tmp/build .
```
Comment (MrChico) [inline circle.yml]:
```suggestion
            cp -r /tmp/build ./build
```

File: Makefile [hunk 2]
```diff
@@ -1,26 +1,21 @@
...
+	@cat build/DepositContract.bin >> deposit_contract.json
 	@/bin/echo -n '"}' >> deposit_contract.json
 
-
-export DAPP_SKIP_BUILD:=1
-export DAPP_SRC:=.
-export DAPP_JSON:=combined.json
-
 test:
+	export DAPP_SKIP_BUILD:=1
```
Comment (MrChico) [inline Makefile]: this does not work:`/bin/sh: line 0: export: `DAPP_SKIP_BUILD:=1': not a valid identifier`. Neither does removing the colon from the `:=`, and I'm not quite sure how to fix it. But just keeping it as before works, so I suggest to keep it like that

Comment (axic) [reply Makefile]: Fixed it.

File: Makefile [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
+	@cat build/DepositContract.bin >> deposit_contract.json
 	@/bin/echo -n '"}' >> deposit_contract.json
 
-
-export DAPP_SKIP_BUILD:=1
-export DAPP_SRC:=.
-export DAPP_JSON:=combined.json
-
 test:
+	export DAPP_SKIP_BUILD=1
```
Comment (MrChico) [inline Makefile]: these environment vars are still not exported to `dapp`

Comment (MrChico) [reply Makefile]: I put them back in the main makefile

**Review by MrChico — APPROVED** [review]

**Pull request merged**